prompts/github.go

```diff
@@ -200,6 +230,21 @@ func ConfigurePublishing(target *workflow.Target, name string) (*workflow.Target
 			},
 		}
 	case "terraform":
+		// Check repository naming convention for Terraform
+		if repo := FindGithubRepository("."); repo != nil {
+			if remoteURL := ParseGithubRemoteURL(repo); remoteURL != "" {
+				// Extract repository name from URL
+				urlParts := strings.Split(remoteURL, "/")
+				if len(urlParts) > 0 {
+					repoName := urlParts[len(urlParts)-1]
+					if err := checkTerraformRepositoryNaming(repoName); err != nil {
+						// Return the target with the warning attached
+						return target, err
```

Comment:
Since this `return` is happening before `target.Publishing` is being set, technically its acting more like an error rather than a warning. I think we'll want to decide what's better or worse here:

- (Current) Not updating the workflow configuration.
- (Alternative) Updating the workflow configuration, then raising the warning.

I _think_ we want the latter by moving this logic below `target.Publishing =`, since its possible for someone to rectify the issue outside the Speakeasy CLI (unless there's other steps necessary).